src/main/java/subway/application/LineService.java

```diff
@@ -2,52 +2,102 @@
 import subway.dao.LineDao;
-import subway.domain.Line;
+import subway.dao.SectionDao;
+import subway.dao.StationDao;
 import subway.dto.LineRequest;
 import subway.dto.LineResponse;
+import subway.dto.LineStationResponse;
+import subway.dto.StationResponse;
+import subway.entity.Line;
+import subway.entity.Section;
+import subway.entity.Station;
 
-import java.util.List;
-import java.util.stream.Collectors;
+import java.util.*;
 
 @Service
 public class LineService {
     private final LineDao lineDao;
+    private final SectionDao sectionDao;
+    private final StationDao stationDao;
 
-    public LineService(LineDao lineDao) {
+    public LineService(final LineDao lineDao, final SectionDao sectionDao, final StationDao stationDao) {
         this.lineDao = lineDao;
+        this.sectionDao = sectionDao;
+        this.stationDao = stationDao;
     }
 
-    public LineResponse saveLine(LineRequest request) {
+    public Long save(LineRequest request) {
         Line persistLine = lineDao.insert(new Line(request.getName(), request.getColor()));
-        return LineResponse.of(persistLine);
+        return persistLine.getId();
     }
 
-    public List<LineResponse> findLineResponses() {
-        List<Line> persistLines = findLines();
-        return persistLines.stream()
-                .map(LineResponse::of)
-                .collect(Collectors.toList());
+    public List<LineStationResponse> findAll() {
+        List<Line> lines = lineDao.findAll();
+        List<Section> sections = sectionDao.findAll();
+        List<Station> stations = stationDao.findAll();
+
+        Map<Long, List<Section>> sectionsByLine = sortSectionsByLine(sections);
+
+        List<LineStationResponse> lineStationResponses = new ArrayList<>();
+        for (Line line : lines) {
+            List<StationResponse> stationResponses = linkStationsByLine(sectionsByLine.get(line.getId()), stations);
+            lineStationResponses.add(LineStationResponse.from(LineResponse.of(line), stationResponses));
+        }
+        return lineStationResponses;
     }
 
-    public List<Line> findLines() {
-        return lineDao.findAll();
+    private Map<Long, List<Section>> sortSectionsByLine(final List<Section> sections) {
+        Map<Long, List<Section>> sectionsByLine = new HashMap<>();
+        for (Section section : sections) {
+            List<Section> sectionEntities = sectionsByLine.getOrDefault(section.getLineId(), new ArrayList<>());
+            sectionEntities.add(section);
+            sectionsByLine.put(section.getLineId(), sectionEntities);
+        }
+        return sectionsByLine;
     }
 
-    public LineResponse findLineResponseById(Long id) {
-        Line persistLine = findLineById(id);
-        return LineResponse.of(persistLine);
+    public LineStationResponse findById(Long id) {
+        Optional<Line> line = lineDao.findById(id);
+        if (line.isEmpty()) {
+            throw new NoSuchElementException("해당하는 ID가 없습니다.");
+        }
```

Comment:
```suggestion
        Line line = lineDao.findById(id)
          .orElseThrow(() -> new NoSuchElementException("해당하는 ID가 없습니다."));
```
find 반환 값이 Optional인 부분은 전부 이렇게 수정할 수 있을 것 같아요.
Optional을 공부해보는 것도 좋은 방법이겠어요.

Reply:
헉 그렇네요! Optional 사용한 부분은 if문 없이도 사용가능하도록 리팩토링하였습니다 ~